sources/packages/api/src/route-controllers/institution-locations/institution-locations.controller.ts

```diff
@@ -67,37 +67,38 @@ export class InstitutionLocationsController extends BaseController {
     }
 
     // If the data is valid the location is saved to SIMS DB.
-    const createdInstitutionlocation =
-      await this.locationService.createtLocation(
+    const createdInstitutionLocation =
+      await this.locationService.createLocation(
         institutionDetails.id,
         dryRunSubmissionResult.data,
       );
 
-    // Save a form to formio to handle the location approval.
+    // Save a form to formIO to handle the location approval.
     const submissionResult = await this.formService.submission(
-      "institutionlocation",
+      "institutionLocation",
       payload,
     );
 
     // Create an application entry on FormFlow.ai, using the
-    // previously created form definition on formio.
+    // previously created form definition on formIO.
     await this.formsFlowService.createApplication({
       formId: submissionResult.formId,
       formUrl: submissionResult.absolutePath,
       submissionId: submissionResult.submissionId,
     });
 
-    return createdInstitutionlocation.id;
+    return createdInstitutionLocation.id;
   }
 
+  @IsInstitutionAdmin()
```

Comment:
The decorators will work in "AND" logic, so this will require admin users and will not let a non-admin have access to the method. When the decorator @HasLocationAccess is present, it will assume also that, if the user is an admin, he will have access to the method.
I would say that we need only the @HasLocationAccess for this method because it is the most restrictive one and also will alow admin to have access to it.

Reply:
institution location should only be updated/edited by admin user right @andrewsignori-aot ?

Reply:
Yes, you are right! So, please remove the other decorator for @HasInstitutionAccess because the @IsAdmin will take precedence.